runtime/lib/ttnn/runtime.cpp

```diff
@@ -89,7 +364,7 @@ Tensor createTensor(std::shared_ptr<void> data,
       createStorage<BorrowedStorage>(data.get(), numElements, dataType),
       ::ttnn::Shape(small_vector_shape), utils::toTTNNDataType(dataType),
       ::ttnn::Layout::ROW_MAJOR);
-  return Tensor(std::static_pointer_cast<void>(tensor), data,
+  return Tensor(std::static_pointer_cast<void>(tensor), nullptr,
```

Comment:
why this change?

Reply:
data shouldn't be referenced for ttnn tensors,there's no lifetime guarantee and we have it in the tensor. Also now that we support device tensors, they're going to be null pointer anyway for these tensors. I kept the member because metal runtime still uses it, but for consistency and safety I'm setting it to null across the board in ttnn runtime

Reply:
But isn't this borrowed data in ttnn? As long as the `Tensor` lives shouldn't we make sure that the `data` pointer lives?

Judging by the name `BorrowedStorage` I would assume that `ttnn` can never deallocate it by itself, since they are not owning it...

Reply:
The data pointer lifetime is managed by the user, we're currently borrowing the pointer with a non-tracking unsafe shared pointer that we create with unsafe_borrow_shared. Therefore having this pointer does not have any implications on lifetime, and for borrowed tensors we can access the data directly from the ttnn::tensor.